api/client/client.go

```diff
@@ -0,0 +1,230 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/textileio/go-textile-core/crypto/symmetric"
+	pb "github.com/textileio/go-textile-threads/api/pb"
+	es "github.com/textileio/go-textile-threads/eventstore"
+	"google.golang.org/grpc"
+)
+
+// Client provides the client api
+type Client struct {
+	client pb.APIClient
+	ctx    context.Context
+}
+
+// NewClient starts the client
+func NewClient(host string, port int) (*Client, error) {
```

Comment:
Shouldn't the `Client` type have a `Close` method that closes the underlying gRPC (and tcp connection)?
Maybe it could also err if there're open txns, or force them to close too and do that type of cleaning stuff.

Reply:
Yea I like that idea. It looks like it could close the grpc connection as well as cancel a context that would cascade down into any requests that are currently in fight (including transactions if we use contexts correctly in that scenario). I'll try to make this change.

Reply:
Ok I added a simple implementation of a client Close method. It closes the grpc connection and calls the root context's cancel function. That root context is used by all client requests. I need to take another pass through with fresh mind/eyes to make sure the root context is correctly used in `Listen`, `ReadTransaction`, and `WriteTransaction`.

Reply:
Issue to track this here https://github.com/textileio/go-textile-threads/issues/141